### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by xguruteam**
This PR implements the dark mode appearance of the app in iOS 13 and higher.

### CODE REVIEW

**Review by xguruteam — COMMENTED** [review]
@piotrgabriel0724 Great jobs!
There is nothing to be changed. The source code is very clean.
Just asking (but no big deal) there is an ambiguous error in `main.storyboard` that exists originally, would you favor to fix it?

**Comment by luccasbeck** [comment]
@piotrgabriel0724 
We've sent three icon sets being used in dark mode.
Did you get it? If so, please insert them into the corresponding icon asset, and update branch.

File: TowerFox/Assets.xcassets/Colors/Folder-label-color.colorset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "color-space" : "srgb",
+        "components" : {
+          "red" : "0xFF",
+          "alpha" : "1.000",
+          "blue" : "0xFF",
+          "green" : "0xFF"
+        }
+      }
+    }
+  ]
```
Comment (xguruteam) [inline TowerFox/Assets.xcassets/Colors/Folder-label-color.colorset/Contents.json]: Is this file used?

File: TowerFox/Assets.xcassets/Colors/Settings-table-head-background.colorset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          "value" : "dark"
+        }
+      ],
+      "color" : {
+        "color-space" : "srgb",
+        "components" : {
+          "red" : "0x1F",
+          "alpha" : "1.000",
+          "blue" : "0x24",
+          "green" : "0x1A"
```
Comment (xguruteam) [inline TowerFox/Assets.xcassets/Colors/Settings-table-head-background.colorset/Contents.json]: change to #0F0F0F

File: TowerFox/Extensions/UIExtensions.swift [hunk 3]
```diff
@@ -303,7 +303,11 @@ extension UIColor {
             blue: CGFloat(arc4random_uniform(256))/255,
             alpha: 1)
     }
-    static let appMainColor = UIColor(rgba: "#465A50")//177FC3
+    static let appMainColor = UIColor(named: "App-primary-color")! //UIColor(rgba: "#465A50")//177FC3
+    static let appOnMainColor = UIColor(named: "App-onprimary-color")!
+    static let appTblHeadBackColor = UIColor(named: "AddProject-view-background")!
+    static let appCollapsTblBackColor = UIColor(named: "Settings-table-head-background")!
+    static let appCollapsTblForeColor = UIColor(named: "Settings-table-head-foreground")!
```
Comment (xguruteam) [inline TowerFox/Extensions/UIExtensions.swift:310]: Main color doesn't work properly.
When I restarted the app after downloading a project, it looked like screenshot;
image
Look at the navigation, weird!

And now, look at this screenshot;
image
It seems that main color wasn't set up correctly.

File: TowerFox/Base.lproj/Main.storyboard [hunk 4]
```diff
@@ -454,7 +454,7 @@
                                                                     </userDefinedRuntimeAttributes>
                                                                 </textField>
                                                             </subviews>
-                                                            <color key="backgroundColor" name="App-onprimary-color"/>
+                                                            <color key="backgroundColor" name="App-textfiled-background"/>
```
Comment (xguruteam) [inline TowerFox/Base.lproj/Main.storyboard:457]: I guess, this is for the bottom button background in review screen,
do not use textfiled-background, add a new asset, and use color for dark mode as #0F0F0F
Also, fix the typo `textfiled-background`

File: TowerFox/Base.lproj/Main.storyboard [hunk 5]
```diff
@@ -2156,7 +2156,7 @@
                                             <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Save and return to list" textAlignment="center" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="iVx-Xc-njN">
                                                 <rect key="frame" x="135" y="7.9999999999999982" width="137.33333333333337" height="16.333333333333329"/>
                                                 <fontDescription key="fontDescription" name="ProximaNovaSoft-Semibold" family="Proxima Nova Soft" pointSize="14"/>
-                                                <color key="textColor" red="0.27450980390000002" green="0.35294117650000001" blue="0.31372549020000001" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+                                                <color key="textColor" name="App-primary-color"/>
```
Comment (xguruteam) [inline TowerFox/Base.lproj/Main.storyboard:2159]: I guess, this is for the bottom button background in review screen,
do not use textfiled-background, add a new asset, and use color for dark mode as #0F0F0F
Also, fix the typo `textfiled-background`

File: TowerFox/Base.lproj/Main.storyboard [hunk 6]
```diff
@@ -2167,7 +2167,7 @@
                                                 </constraints>
                                             </imageView>
                                         </subviews>
-                                        <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
+                                        <color key="backgroundColor" name="App-textfiled-background"/>
```
Comment (xguruteam) [inline TowerFox/Base.lproj/Main.storyboard]: I guess, this is for the bottom button background in review screen,
do not use textfiled-background, add a new asset, and use color for dark mode as #0F0F0F
Also, fix the typo `textfiled-background`

**Review by xguruteam — COMMENTED** [review]
@piotrgabriel0724 
Please see the new comments for the final fixes.
And please add dark mode color for `App-small-btn-background` as #474747

File: TowerFox/Extensions/UIExtensions.swift (hunk 3 above)
Comment (xguruteam) [reply TowerFox/Extensions/UIExtensions.swift:310]: @piotrgabriel0724 
I can't see any changes for it, would you like to check any of missing files in there?

**Review by xguruteam — COMMENTED** [review]
@piotrgabriel0724 
Well done, everything looks good.
I will let you know client feedback at soon.

**Comment by piotrgabriel0724** [comment]
Thank you!

**Pull request merged**